src/Product.php

```diff
@@ -214,17 +219,43 @@ public static function display_store_stock_status_block() {
     }
 
     global $product;
+    $store_stocks = static::getStockByStore($product);
 
-    $stocks = static::getStockByStore($product);
-    $raw = $stocks;
-    foreach ($stocks as $name => $types) {
+    $availability = [];
+    if (!empty(static::$allVariationsStock) && count(static::$allVariationsStock) > 1) {
+      foreach (reset(static::$allVariationsStock) as $location => $stocks) {
+        $availability[$location] = [];
+        foreach ($stocks as $type => $stock) {
+          if (!isset($availability[$location][$type])) {
+            $availability[$location][$type] = [];
+          }
+        }
+      }
+      foreach (static::$allVariationsStock as $product_id => $locations) {
+        foreach ($locations as $location => $stocks) {
+          foreach ($stocks as $type => $stock) {
+            if ($stock != 'none' && $type == 'showroom') {
+              $availability[$location][$type][] = $product_id;
+              $store_stocks[$location][$type] = 'high-asterisk';
+            }
+          }
+        }
+      }
+    }
```

Comment:
```suggestion
    if (!empty(static::$allVariationsStock) && count(static::$allVariationsStock) > 1) {
      foreach (reset(static::$allVariationsStock) as $location => $stocks) {
        $availability[$location] = [];
        foreach ($stocks as $type => $stock) {
          if (!isset($availability[$location][$type])) {
            $availability[$location][$type] = [];
          }
        }
      }
      foreach (static::$allVariationsStock as $product_id => $locations) {
        foreach ($locations as $location => $stocks) {
          foreach ($stocks as $type => $stock) {
            if ($stock != 'none' && $type == 'showroom') {
              $availability[$location][$type][] = $product_id;
              $store_stocks[$location][$type] = 'high-asterisk';
            }
          }
        }
      }
    }
```

Is there any way to refactor this in a different method(s) including documentation? 3 levels of foreach is really hard to read.

Reply:
Updated in the last commit.